### PR TITLE
Update css-and-tailwind.mdx

### PR DESCRIPTION
**Pull request by clsty**
#### Description

- What does this PR change? Give us a brief description.

Add an example showing how to do auto-numbering for headings using css.

- Did you change something visual? A before/after screenshot can be helpful.

Take this markdown as example:
```markdown
---
title: This is a Page Title
---

Hi.

## Heading(0.1)
## Heading(0.2)
### Heading(0.2.1)
### Heading(0.2.2)
## Heading(0.3)

# Heading(1)
## Heading(1.1)
## Heading(1.2)
### Heading(1.2.1)
### Heading(1.2.2)
## Heading(1.3)

# Heading(2)
```

Before:

Before

After (applying the documented example in css file manually):

After

### CODE REVIEW

**Comment by clsty** [comment]
> Hi @clsty! Thank you for opening a PR and wanting to contribute to Starlight’s docs.
> 
> In general, in the Starlight docs we only document techniques and behaviours that are specific to Starlight and not general concepts in CSS or HTML etc. like this example.
> 
> We do however have a collection of links to community recipes, so if you feel like publishing a guide to this technique on a personal blog or a platform like Dev.to, we’d be happy to link to that: https://starlight.astro.build/resources/community-content/#recipes-and-guides
> 
> In the future, please always feel free to open [a discussion](https://github.com/withastro/starlight/discussions) before starting work on a PR to see if your idea is something that might make sense as part of Starlight or is better as external content. We’re always happy to provide guidance.

Thanks for quick response.

However, excuse me but I really can't agree with the statement above.
If it was a "general" thing of CSS, then [this post](https://stackoverflow.com/a/43625080) should just have solved my problem.
However it did NOT, the output using that is awful and full of problems, and I have to figure out why, spent hours to debug and finally find out a flawless method.

The point is that starlight/astro renders page title as `h1` element, which other markdown renders usually does NOT (otherwise [this post](https://stackoverflow.com/a/43625080) will not be a general solution).

In conclusion, the example here is ONLY suitable for starlight/astro.
I hope you could reconsider this PR.

**Comment by delucis** [comment]
While the code is tweaked for Starlight, what I meant is that the general approach is not — there are many many custom CSS approaches that users might want and it is not feasible for us to include every single one in our docs, where we are responsible for maintaining them, checking them when we change code etc. I agree that this might be useful and helpful though and would love to link to a guide that shows this if you’d like to publish one!

Regarding your code, instead of complicating things with the exceptions, you probably want to target all your styles to just the Markdown content:

```css
.sl-markdown-content h1 {
    counter-increment: h1;
    counter-reset: h2;
}

.sl-markdown-content h2 {
    counter-increment: h2;
    counter-reset: h3
}

.sl-markdown-content h3 {
    counter-increment: h3;
    counter-reset: h4
}

.sl-markdown-content h4 {
    counter-increment: h4;
}

.sl-markdown-content h1:before {
    content: counter(h1) ". "
}

.sl-markdown-content h2:before {
    content: counter(h1) "." counter(h2) ". "
}

.sl-markdown-content h3:before {
    content: counter(h1) "." counter(h2) "." counter(h3) ". "
}

.sl-markdown-content h4:before {
    content: counter(h1) "." counter(h2) "." counter(h3) "." counter(h4) ". "
}
```
